### ISSUE TITLE
iOS: Missing CFBundleIconName and asset catalog

### ISSUE THREAD

**Issue by MagicalTux** [fyne-io/fyne#1504]
### Describe the bug:

When building for iOS and deploying an app, an error is returned by Apple:

    ITMS-90713: Missing Info.plist value - A value for the Info.plist key 'CFBundleIconName' is missing in the bundle '...'. Apps built with iOS 11 or later SDK must supply app icons in an asset catalog and must also provide a value for this Info.plist key. For more information see http://help.apple.com/xcode/mac/current/#/dev10510b1f7.

This issue was found while working on issue #1497 

### To Reproduce:
Steps to reproduce the behaviour:
1. Build for iOS
2. Deploy
3. Email from Apple will arrive with the error message

### Device (please complete the following information):
 - OS: MacOS
 - Version: 10.15.7
 - Go version: 1.15.3
 - Fyne version: 2d9a597

**Comment by MagicalTux** [comment]
Looking at another project the following is found:

`project.pbxproj` file contains the following line under `buildSettings`:

    ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;

A folder `Assets.xcassets/AppIcon.appiconset` exists containing:

* Icon png files in various sizes
* `Contents.json` file:

```json
{
  "images" : [
    {
      "size" : "20x20",
      "idiom" : "iphone",
      "filename" : "Icon-App-20x20@2x.png",
      "scale" : "2x"
    },
    ....
  ],
  "info" : {
    "version" : 1,
    "author" : "xcode"
  }
}
```

**Comment by MagicalTux** [comment]
Additionally, found in `project.pbxproj` under `objects`:

    97C146FE1CF9000F007C117D /* Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 97C146FD1CF9000F007C117D /* Assets.xcassets */; };
    ...
    97C146FD1CF9000F007C117D /* Assets.xcassets */ = {isa = PBXFileReference; lastKnownFileType = folder.assetcatalog; path = Assets.xcassets; sourceTree = "<group>"; };

under `children`:

    97C146FD1CF9000F007C117D /* Assets.xcassets */,

**Comment by andydotxyz** [comment]
This is great help thanks. I guess we'd need to make sure that the idiom and scale are set correctly for 4 required icon files that we bundle?

**Comment by MagicalTux** [comment]
The following idiom/sizes are present:

 * iphone 20x20 2x
 * iphone 20x20 3x
 * iphone 29x29 1x
 * iphone 29x29 2x
 * iphone 29x29 3x
 * iphone 40x40 2x
 * iphone 40x40 3x
 * iphone 60x60 2x
 * iphone 60x60 3x
 * ipad 20x20 1x
 * ipad 20x20 2x
 * ipad 29x29 1x
 * ipad 29x29 2x
 * ipad 40x40 1x
 * ipad 40x40 2x
 * ipad 76x76 1x
 * ipad 76x76 2x
 * ipad 83.5x83.5 2x
 * ios-marketing 1024x1024 1x

The png file has the scale applied (for example 20x20@2x will be a 40x40 png file)

**Comment by andydotxyz** [comment]
These hints are cracking thanks so much @MagicalTux - I have a proof of concept asset compiler locally and running through the various verification / upload steps now.

**Comment by andydotxyz** [comment]
OK! I hit some bitcode issues, but this is now working :)

<img width="852" alt="Screenshot 2020-11-15 at 09 03 36" src="https://user-images.githubusercontent.com/294436/99180835-9b6dcb00-2721-11eb-801b-b0a2d6758a67.png">

**Comment by andydotxyz** [comment]
Merged in `release/v1.4.x` ready for next point release :)
